packages/astro/src/@types/astro.ts

```diff
@@ -1497,66 +1497,80 @@ export interface AstroUserConfig {
-			 * will redirect to `/[defaultLocale]` when `prefixDefaultLocale: true` is set.
-			 *
-			 * Set `redirectToDefaultLocale: false` to disable this automatic redirection at the root of your site:
-			 * ```js
-			 * // astro.config.mjs
-			 * export default defineConfig({
-			 *   i18n:{
-			 *     defaultLocale: "en",
-			 * 		locales: ["en", "fr"],
-			 *     routing: {
-			 *       prefixDefaultLocale: true,
-			 *       redirectToDefaultLocale: false
-			 *     }
-			 *   }
-			 * })
-			 *```
-			 * */
-			redirectToDefaultLocale?: boolean;
-
-			/**
-			 * @name i18n.routing.strategy
-			 * @type {"pathname"}
-			 * @default `"pathname"`
-			 * @version 3.7.0
-			 * @description
-			 *
-			 * - `"pathname": The strategy is applied to the pathname of the URLs
+			 * You will be responsible for writing your own routing logic, or executing Astro's i18n middleware manually alongside your own.
 			 */
-			strategy?: 'pathname';
-		};
+		 	'manual'
+			| {
+					/**
+					 * @docs
+					 * @name i18n.routing.prefixDefaultLocale
+					 * @kind h4
+					 * @type {boolean}
+					 * @default `false`
+					 * @version 3.7.0
+					 * @description
+					 *
+					 * When `false`, only non-default languages will display a language prefix.
+					 * The `defaultLocale` will not show a language prefix and content files do not exist in a localized folder.
+					 *  URLs will be of the form `example.com/[locale]/content/` for all non-default languages, but `example.com/content/` for the default locale.
+					 *
+					 * When `true`, all URLs will display a language prefix.
+					 * URLs will be of the form `example.com/[locale]/content/` for every route, including the default language.
+					 * Localized folders are used for every language, including the default.
+					 */
+					prefixDefaultLocale?: boolean;
+
+					/**
+					 * @docs
+					 * @name i18n.routing.redirectToDefaultLocale
+					 * @kind h4
+					 * @type {boolean}
+					 * @default `true`
+					 * @version 4.2.0
+					 * @description
+					 *
+					 * Configures whether or not the home URL (`/`) generated by `src/pages/index.astro`
+					 * will redirect to `/[defaultLocale]` when `prefixDefaultLocale: true` is set.
+					 *
+					 * Set `redirectToDefaultLocale: false` to disable this automatic redirection at the root of your site:
+					 * ```js
+					 * // astro.config.mjs
+					 * export default defineConfig({
+					 *   i18n:{
+					 *     defaultLocale: "en",
+					 * 		locales: ["en", "fr"],
+					 *     routing: {
+					 *       prefixDefaultLocale: true,
+					 *       redirectToDefaultLocale: false
+					 *     }
+					 *   }
+					 * })
+					 *```
+					 * */
+					redirectToDefaultLocale?: boolean;
+
+					/**
+					 * @name i18n.routing.strategy
+					 * @type {"pathname"}
+					 * @default `"pathname"`
+					 * @version 3.7.0
+					 * @description
+					 *
+					 * - `"pathname": The strategy is applied to the pathname of the URLs
+					 */
+					strategy?: 'pathname';
+			  };
```

Comment:
I can't make the suggestion in this file, but **line 1609** should also be updated to remove links to the guide page:

```
Both page routes built and URLs returned by the `astro:i18n` helper functions [`getAbsoluteLocaleUrl()`](https://docs.astro.build/en/reference/api-reference/#getabsolutelocaleurl) and [`getAbsoluteLocaleUrlList()`](https://docs.astro.build/en/reference/api-reference/#getabsolutelocaleurllist) will use the options set in `i18n.domains`.
```